upup/pkg/fi/cloudup/awstasks/sshkey.go

```diff
@@ -108,6 +109,13 @@ func (e *SSHKey) Run(c *fi.Context) error {
 		}
 		glog.V(2).Infof("Computed SSH key fingerprint as %q", keyFingerprint)
 		e.KeyFingerprint = &keyFingerprint
+	} else if e.PublicKey == nil && c.Cluster.Spec.SSHKeyName != "" {
```

Comment:
So in the tasks layer, we try not to refer back to the Cluster Spec; it's sort of crossing the layers - you might think of the tasks layer as being basically terraform.  We probably should rename the `Name` field to be `SSHKeyName` in the task, but I think for the logic here you just want to check `else if e.PublicKey == nil && e.Name != ""`